source/D2Common/src/Drlg/DrlgOutPlace.cpp

```diff
@@ -1917,972 +1922,661 @@ void __fastcall sub_6FD82750(D2DrlgStrc* pDrlg, int nStartId, int nEndId)
+			nMaxX = pCurrentJungle->pDrlgCoord.nWidth + pCurrentJungle->pDrlgCoord.nPosX;
+		// Original game does not change nMinY: error or just never happens ?
+	}
+	// Note: Original game uses FOG_10025 not FOG_Assertion
+	D2_ASSERT((nMaxX - nMinX) % nPresetBlocksDimensions == 0);
+	D2_ASSERT((nMaxY - nMinY) % nPresetBlocksDimensions == 0);
+
+
+	const int32_t nPresetsWidth = (nMaxX - nMinX) / nPresetBlocksDimensions + 2;
+	const int32_t nPresetsHeight = (nMaxY - nMinY) / nPresetBlocksDimensions + 2;
+	const int32_t nPresets = nPresetsWidth * nPresetsHeight;
+
+	const size_t nPresetsAllocSize = nPresets * sizeof(uint32_t);
+	int32_t* pPreset0 = (int32_t*)D2_ALLOC_SERVER(pDrlg->pMempool, nPresetsAllocSize);
+	int32_t* pPreset1 = (int32_t*)D2_ALLOC_SERVER(pDrlg->pMempool, nPresetsAllocSize); // Could be some kind of distance field ?
+	int32_t* pPreset2 = (int32_t*)D2_ALLOC_SERVER(pDrlg->pMempool, nPresetsAllocSize);
+	int32_t* pLevelPresetId = (int32_t*)D2_ALLOC_SERVER(pDrlg->pMempool, nPresetsAllocSize);
+
+	do {
+		DRLG_GenerateJunglesAttachPoints(pDrlg, tJungles,
+			nMinX, nMinY,
+			nSpiderForestLevelSizeX, nSpiderForestLevelSizeY,
+			nPresetsWidth, nPresetsHeight,
+			pPreset0, pPreset1, pPreset2, pLevelPresetId);
+		DRLG_JungleComputeConnexity(pDrlg, nPresetsWidth, nPresetsHeight, pPreset0, pPreset1, pPreset2, pLevelPresetId);
+	} while (!DRLG_JungleUpdateAttachPointsDirections(pDrlg, nPresetsWidth, nPresetsHeight, pPreset0, pPreset2, pLevelPresetId));
+
+
+	for (int32_t nCurrentPresetY = 0; nCurrentPresetY < nPresetsHeight; nCurrentPresetY++)
+	{
+		for (int32_t nCurrentPresetX = 0; nCurrentPresetX < nPresetsWidth; nCurrentPresetX++)
+		{
+			const int32_t nCurrentPresetIndex = nCurrentPresetY * nPresetsWidth + nCurrentPresetX;
+			pLevelPresetId[nCurrentPresetIndex] = DRLG_JungleNormalizeLevelPresetId(pLevelPresetId[nCurrentPresetIndex]);
+		}
+	}
+
+	for (int nJungleIdx = 0; nJungleIdx < JUNGLE_MAX_ATTACH; ++nJungleIdx)
+	{
+		D2JungleStrc& tCurrentJungle = tJungles[nJungleIdx];
+		const int32_t nSpiderForestLevelPresetsBlocksSizeX = nSpiderForestLevelSizeX / nPresetBlocksDimensions;
+		const int32_t nSpiderForestLevelPresetsBlocksSizeY = nSpiderForestLevelSizeY / nPresetBlocksDimensions;
+
+		int nJungleDefsIndex = 0;
+		for (int nBlockY = 0; nBlockY < nSpiderForestLevelPresetsBlocksSizeY; ++nBlockY)
+		{
+			for (int nBlockX = 0; nBlockX < nSpiderForestLevelPresetsBlocksSizeX; ++nBlockX)
+			{
+				const int32_t nCurrentPresetIndex = nBlockX + tCurrentJungle.nPresetsBlocksX + nPresetsWidth * (nBlockY + tCurrentJungle.nPresetsBlocksY);
+				const int32_t nLevelPresetId = pLevelPresetId[nCurrentPresetIndex];
+				tCurrentJungle.pJungleDefs[nJungleDefsIndex] = nLevelPresetId;
+				if (nLevelPresetId > LVLPREST_ACT3_JUNGLE_TAIL)
+					++tCurrentJungle.nJungleDefs;
+				++nJungleDefsIndex;
+			}
+		}
+	}
+
+	// Bubble sort on Y position
+	bool bSorted = false;
+	while (bSorted);
```

Comment:
Something is wrong here. Either this needs to be a `do while` loop or `bSorted` has to be set to `true` before. Also, it of course has to be `while(bSorted)` instead of `while(bSorted);` then.

I wonder if you could just use `std::sort` here instead. Maybe that's the best option anyway

Reply:
Well spotted, funnily enough this wasn't causing issues because the sort always finished in one iteration. Fixing.